src/errors.ts

```diff
@@ -80,6 +80,24 @@ export class RelayerError extends BobError {
     }
 }
 
+export class DelegatedProverError extends BobError {
+    public code: number;
+    constructor(code: number, message: string) {
+        super(`Delegated prover response incorrect (code ${code}): ${message}`);
+        this.code = code;
+    }
+}
+
+export class ServiceError extends BobError {
+    public code: number;
+    public message: string;
+    constructor(code: number, message: string) {
+        super(`Service response incorrect (code ${code}): ${message}`);
+        this.code = code;
+        this.message = message
+    }
```

Comment:
The standard `Error` JS object already contains  the `message` field. When you call `super(str)` in your constructor the message of the error is initialized with `str` So in your code that line `this.message = message` will override the formatted message from that line `super(`Service response incorrect (code ${code}): ${message}`);` If you want to hold response message separately you should rename local message field:

```suggestion
    public responseText: string;
    constructor(code: number, message: string) {
        super(`Service response incorrect (code ${code}): ${message}`);
        this.code = code;
        this.responseText = message
    }
```

Or just remove it at all:

```suggestion
    constructor(code: number, message: string) {
        super(`Service response incorrect (code ${code}): ${message}`);
        this.code = code;
    }
```

Reply:
Fixed in https://github.com/zkBob/zkbob-client-js/pull/87/commits/5dda91709f1d737dac43b8b9dba6bc76d3dce524